pkg/model/crd.go

```diff
@@ -657,6 +657,45 @@ func (r *CRD) GetAllRenames(op OpType) (map[string]string, error) {
 	return renames, nil
 }
 
+// FindIdentifierInShape returns true and the resource's identifier iff shape
+// contains a matching field or corresponding plural meaning it can be populated
+// using the resource identifier.
+func (r *CRD) FindIdentifierInShape(
+	shape *awssdkmodel.Shape) (string, bool) {
+	resIdentifier := ""
+	found := false
+	resourceIdentifiers := []string{
+		"Name",
+		"Names",
+		r.Names.Original + "Name",
+		r.Names.Original + "Names",
+		r.Names.Original + "Id",
+		r.Names.Original + "Ids",
+	}
+
+	// Locate an identifier within resource
+	for _, id := range resourceIdentifiers {
+		_, found := r.SpecFields[id]
+		if !found {
+			_, found = r.StatusFields[id]
+		}
+		if found {
+			resIdentifier = id
+			break
+		}
+	}
```

Comment:
Not a big deal but we could also exit early if !found in `SpecFields` and !found in `StatusFields`